http/src/main/scala/org/broadinstitute/dsde/workbench/leonardo/http/service/LeoAppServiceInterp.scala

```diff
@@ -868,7 +878,9 @@ final class LeoAppServiceInterp[F[_]: Parallel](config: AppServiceConfig,
       cloudContext = cloudContext,
       clusterName = clusterName,
       location = config.leoKubernetesConfig.clusterConfig.location,
-      region = config.leoKubernetesConfig.clusterConfig.region,
+      region =
+        if (cloudContext.cloudProvider == CloudProvider.Azure) RegionName(regionName)
```

Comment:
Isn't the `RegionName` class for Google regions? Should this conditional be a `!=`?

Reply:
region is a RegionName

`case class SaveKubernetesCluster(cloudContext: CloudContext,
                                 clusterName: KubernetesClusterName,
                                 location: Location,
                                 region: RegionName,
                                 status: KubernetesClusterStatus,
                                 ingressChart: Chart,
                                 auditInfo: AuditInfo,
                                 defaultNodepool: DefaultNodepool // Question: does this have to be `DefaultNodepool`?
)

Reply:
Right, but that's a legacy of the `region` field up to now being populated only for GKE apps, right? And now we're adding support for Azure, which would have its own region syntax, I thought

Reply:
you could define the region variable as for SaveKubernetesCluster as Either[RegionName, Region], with region being the Azure region, then change the file that defines the cluster component to reflect that (org/broadinstitute/dsde/workbench/leonardo/db/KubernetesClusterComponent.scala)

Reply:
then region for GCP apps would be `Left(config.leoKubernetesConfig.clusterConfig.region)` and for Azure apps `Right(regionName)`

Reply:
but that might be tricky on the decoder side...

Reply:
I'd be _ok_ with leaving this as is (there's similar logic in runtimeV2ServiceInterp with ZoneName) as long as we leave a comment explaining

Reply:
But the regions aren't the same values, are they? How is it going to work if we're trying to persist Azure regions as Google regions?

Reply:
I guess they both have code for persisting arbitrary strings as regions. But doesn't that open us up to problems down the line?

Reply:
I guess we could add a new RegionName(value: String) class to the kubernetesModels that would be cloud agnostic

Reply:
It is just a string in the database. So any decoding could be done based on cloud type of azure or google to find the label or valid string based on the strings in Region vs RegionName.

Reply:
@lucymcnatt @mlilynolting - would you have time for a quick meeting to discuss the best practice for RegionName and Region? I think @lucymcnatt's suggestion of having a base RegionName class in kubernetesModels sounds like the right approach, but I need to look into how or if our DAO handles polymorphism and how the contracts are using RegionName. How about 11am?

Reply:
Sure! I have a 1:1 at 11 but I'm free between 12 and 2; I can check in if you and Lucy end up meeting earlier

Reply:
the way we both overload AND duck-type our data model gives me the heebie-jeebies👻